src/pages/dev/axelarjs-sdk/axelar-query-api.mdx

```diff
@@ -26,15 +26,21 @@ This is a useful query for GMP transactions, when invoking `callContract` or `ca
 
 IMPORTANT NOTES:
 
-* You must ensure the correct `gasLimit` is used for your gas estimation. The `gasLimit` parameter can be used here to override the default 700K gas limit that is used. 
+* You must ensure the correct `gasLimit` is used for your gas estimation. The gas limit is equal to the execution gas cost on the destination chain. You can test out GMP calls to find the correct value.
 * The `minGasPrice` and `gasMultipler` parameters can be adjusted to enhance execution certainty of cross-chain transactions.
-* The response to the method gives a single value (in `wei`) of the total estimated gas required for the cross-chain call. In the GMPParams parameter, there is an option to show a full breakdown of fees instead by specifying `true` for `showDetailedFees`. If you do this, you will have to add the individual components of the fees yourself, 
-  e.g. `(baseFee + executionFeeWithMultiplier)` for regular way GMP calls, or `(baseFee + executionFeeWithMultiplier + expressFee)` for GMP Express calls
+* L2 chains. For execution of GMP calls where the destination chain is an EVM L2 chain, it is important to note that all L2 chains have an extra cost of posting the executed transaction back to the L1 chain. This `estimateGasFee` SDK method will aggregate that estimated cost for you. The parameters to call out specifically for L2s are:
+  - the `gasLimit` parameter, which are the actual gas units required to execute the tx on the L2 chain. You can check this empirically by looking at the gas used to execute the `AxelarExecutable` on the destination chain.
```

Comment:
Where can devs find the `gasLimit` they need to pass in here for the execution fee? Could be helpful to call this out.

Reply:
they should know the gas limit of the calls to the executables they are calling on the destination chain, like how much gas a contract call should take. they can also look this up empirically for any call on axelarscan,
e.g. for this tx: https://axelarscan.io/gmp/0x604dc7aaa8ac5efbacc491a693662bfce503008ab62e74738fa879a48e0a5195:5
the gas on BNB here (https://bscscan.com/tx/0x93cd11e8f189fd983a94e9df6af4e0b5776c4f04d2a75cbd234a6ad35d4a195d) was 369K